### ISSUE TITLE
Property Injection changed with recent Typescript versions.

### ISSUE THREAD

**Issue by danelowe** [thiagobustamante/typescript-ioc#90]
I was just having a look into this library and came across an issue with recent Typescript (4.3.5)

TLDR - Property injection needs the `declare` keyword. (or `useDefineForClassFields`set to false in tsconfig)

```
@Inject
private declare helloService: HelloService;
```

I can look at submitting a PR for changing the readme if I get some time and look further into this library.

The issue is described well in the [Typescript 3.7 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier).

The `useDefineForClassFields` now defaults to true as of Typescript 4.x. I'd be inclined to leave it as true based on their description of it being the most likely standard going forward.
